ext/js/comm/api.js

```diff
@@ -375,13 +375,15 @@ export class API {
         const data = {action, params};
         return new Promise((resolve, reject) => {
             try {
-                this._yomitan.sendMessage(data, (response) => {
-                    this._checkLastError(chrome.runtime.lastError);
+                this._webExtension.sendMessage(data, (response) => {
+                    this._webExtension.getLastError();
```

Comment:
`chrome.runtime.lastError` confuses me due to [this behavior](https://developer.chrome.com/docs/extensions/reference/api/runtime#property-lastError):

 > This is only defined within the scope of that function's callback. If an error is produced, but runtime.lastError is not accessed within the callback, a message is logged to the console listing the API function that produced the error. 

I guess your refactor still keeps it "within the scope of that function's callback" even through there is some indirection added. Is that understanding correct? 

(I was always confused why `_checkLastError` didn't just check `chrome.runtime.lastError` internally and assumed it was due to some restriction regarding the above behavior...)

Reply:
It works so long as it is within the invocation of the callback, the stack trace doesn't matter. This is similar to how some web APIs can only be invoked (without error) in an event handler which was initiated from a user gesture. Anything inside the invocation of the handler is fine (async is not, since it's deferred).

https://html.spec.whatwg.org/multipage/interaction.html#user-activation-processing-model

The _checkLastError was just a dumb way of doing it. I'm pretty sure that I could have just written `void chrome.runtime.lastError;` to make the property be read and then thrown away, but I didn't think of it at the time. I also forget whether or not that actually _does_ work, or if that is another one of those things that is no-op'd out which effectively results in the property not being read.